src/utils/validationService.ts

```diff
@@ -0,0 +1,58 @@
+import InvalidIdentifierError from '../errors/InvalidIdentifierError';
+import ParametersError from '../errors/ParametersError';
+import ResultsEvent from '../interfaces/ResultsEvent';
+import VehicleEvent from '../interfaces/VehicleEvent';
+
+const basicRegex = /^[a-zA-Z0-9]+$/;
+
+const validateVehicleEvent = (event: VehicleEvent): boolean => {
+  if (!event.VehicleRegMark && !event.vinNumber && !event.trailerId) {
+    throw new ParametersError('No parameter defined');
+  }
+
+  if (
+    (event.VehicleRegMark !== undefined && event.vinNumber !== undefined)
```

Comment:
tests are really good and we have some nice coverage. Just as a FYI I think 2 conditions for the validation service are not covered so I ll leave it to you in case we want to add them

```
event.trailerId !== undefined
event.testnumber !== undefined
```

Reply:
I think I've got all the combinations, but I have just dug up a couple of badly named tests, so cheers :)